packages/grant-explorer/src/features/collection/useCollections.tsx

```diff
@@ -0,0 +1,46 @@
+import { useCallback, useState } from "react";
+import { Project } from "../api/types";
+import { CollectionsModal, NewCollectionsModal } from "./CollectionsModal";
+
+enum CurrentModal {
+  NONE,
+  ALLCOLLECTIONS,
+  NEWCOLLECTIONS,
+}
+
+const useCollections = () => {
+  const [activeProject, setActiveProject] = useState<Project | null>(null);
+  const [currentModal, setCurrentModal] = useState<CurrentModal>(
+    CurrentModal.NONE
+  );
+
+  const openModal = useCallback((project: Project) => {
+    setActiveProject(project);
+    setCurrentModal(CurrentModal.ALLCOLLECTIONS);
+  }, []);
+
+  const closeModal = useCallback(() => {
+    setCurrentModal(CurrentModal.NONE);
+    setActiveProject(null);
+  }, []);
+
+  const renderModal = () => (
+    <>
+      <CollectionsModal
+        project={activeProject}
+        open={currentModal === CurrentModal.ALLCOLLECTIONS}
+        createNewCollection={() => setCurrentModal(CurrentModal.NEWCOLLECTIONS)}
+        closeModal={closeModal}
+      />
+      <NewCollectionsModal
+        project={activeProject}
+        open={currentModal === CurrentModal.NEWCOLLECTIONS}
+        closeModal={closeModal}
+      />
+    </>
+  );
```

Comment:
Haven't seen this pattern before, is it common to render components from a hook? Why not put this in a component and conditionally render if currentModal is set?